src/services/DynamoDBService.ts

```diff
@@ -180,19 +182,34 @@ export class DynamoDBService {
      * @param item - the item to be inserted or updated during the transaciton.
      * @param oldItem - the current item that already exists in the database.
      */
-    public transactWrite(item: any, oldItem?: any): Promise<PromiseResult<DocumentClient.TransactWriteItemsOutput, AWS.AWSError>> {
-        const query: DocumentClient.TransactWriteItemsInput = { TransactItems: [
-            {
-                Put: {
-                    TableName: this.tableName,
-                    Item: item,
-                    ConditionExpression: "testNumber = :OldTestNumber",
-                    ExpressionAttributeValues: {
-                        ":OldTestNumber": oldItem.testNumber
+    public transactWrite(item: any, oldItem?: any, numberType: string = NUMBER_TYPE.TEST_NUMBER): Promise<PromiseResult<DocumentClient.TransactWriteItemsOutput, AWS.AWSError>> {
```

Comment:
A common `transactWrite` which takes `ConditionExpression` as a parameter. that will avoid conditional logic.